test/topology_custom/test_table_desc_read_barrier.py

```diff
@@ -55,7 +55,15 @@ async def test_table_desc_read_barrier(request, manager: ManagerClient) -> None:
     async with inject_error(manager.api, servers[0].ip_addr, 'group0_state_machine::delay_apply'):
         logger.info("Altering table")
         await cql.run_async("alter table ks.t add s1 int", host=hosts[1])
-        # TODO: trigger the read barrier once available
-        desc_schema = [await cql.run_async("DESC SCHEMA", host=h) for h in hosts]
 
+        # verify that there is a schema difference
+        # (raft apply delayed, so the first server doesn't have the new schema yet)
+        desc_schema = [await cql.run_async("DESC SCHEMA", host=h) for h in hosts]
```

Comment:
This test has the potential of becoming flaky if the sleep (I think I remember it is 500 milliseconds) ends before this code gets to run - perhaps because of some super-slow test machine.

I'm not sure what to do about this. Ideally instead of a sleep injection you'll have injection that "hangs" forever and a second injection which releases this hang. But maybe it's not easy to do.

Reply:
Yes, I was thinking about it as well, the tricky part is that the wait would need to be unblocked after the read barrier is triggered and waiting (so basically a deadlocking scenario) ... when testing locally with 100 reps and pool size 10 and 20 the test was always 100% passing (in case of flackiness we'd probably need to increase the timeout).

Will still give it a thought, if I can see some way to make it work fully deterministically.

Reply:
I managed to use handler message to resume after calling the read barrier and still avoid the deadlock, by using the `asyncio.gather()` to unpause in parallel. There is still little bit if non-determinism in the sense that the resume might happen before the read barrier (added an extra small sleep before resume to make that less likely), but the test will still succeed in both cases (thus it should not be flaky in any case).